frontend/model/contracts/shared/time.js

```diff
@@ -132,13 +132,17 @@ export function humanDate (
 }
 
 export function isPeriodStamp (arg: string): boolean {
-  return /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(arg)
+  return isIsoString(arg)
 }
 
 export function isFullMonthstamp (arg: string): boolean {
   return /^\d{4}-(0[1-9]|1[0-2])$/.test(arg)
 }
 
+export function isIsoString (arg: string): boolean {
```

Comment:
We could, and I've quickly considered it before writing this function, but I would advise slightly against it

- Even though remembering and respecting the correct letter-casing of the various names is not that hard, it's easier to just camel-case everything.
- Too many uppercase acronyms can quickly become ugly and add visual clutter, except maybe in suffix position. For example `JSONRPCCall` isn't that readable. Even with just one acronym per name like in `JSONData`, if that name is used several times in a block it will add more clutter than `jsonData`
- Uppercase acronyms will yield names easily confused with constructor names because of a starting uppercase letter (that's not the case here, but well in declarations like `let ISOString = '...Z'`. We likely don't want that).

Reply:
@taoeffect 
> For example, periodStampGivenDate iterates backwards in the periods, but it could be calling this getter instead to that, couldn't it?

True, but please note that doing so might make removing `periodAfter/beforePeriod` in the future more difficult